packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -214,6 +215,42 @@ export class CacheService {
     }
   }
 
+  /**
+   * Sets multiple values in the cache, each associated with its respective key.
+   * If the shared cache is enabled and an error occurs while setting in it,
+   * the internal cache is used as a fallback.
+   * @param {Record<string, any>} entries - An object containing key-value pairs to cache.
+   * @param {string} callingMethod - The name of the method calling the cache.
+   * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
+   * @param {boolean} shared - Whether to use the shared cache (optional, default: false).
+   */
+  public mSet(
+    entries: Record<string, any>,
+    callingMethod: string,
+    requestIdPrefix?: string,
+    shared: boolean = false,
+  ): void {
+    if (shared && this.isSharedCacheEnabled) {
+      try {
+        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.REDIS, CacheService.methods.MSET).inc(1);
+        this.sharedCache.mSet(entries, callingMethod, requestIdPrefix);
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(
+          `${requestIdPrefix} Error occurred while setting the cache to Redis. Fallback to internal cache. Error is: ${redisError.fullError}`,
+        );
+        // Fallback to internal cache for each failed key-value pair individually
+        for (const [key, value] of Object.entries(entries)) {
+          this.internalCache.set(key, value, callingMethod, undefined, requestIdPrefix);
```

Comment:
Why fall back to individual set vs multiSet?

Reply:
I guess I worked by parts and forgot to changed it.

Thanks for noticing!!

Reply:
✅ Done